altair_pandas/_core.py

```diff
@@ -230,6 +230,12 @@ def scatter(self, x, y, c=None, s=None, **kwargs):
         if s is not None:
             encodings["size"] = _valid_column(s)
         columns = list(set(encodings.values()))
+        if kwargs.get("colormap"):
+            if c is None:
+                raise ValueError("'c' should be defined for using 'colormap'")
+            encodings["color"] = alt.Color(
+                encodings["color"], scale=alt.Scale(scheme=kwargs.get("colormap"))
+            )
```

Comment:
Let's put this in the ``if c is not None`` block above, so that colormap just has no effect unless ``c`` is defined.

Reply:
I was doing so, to avoid touching `columns = list(set(encodings.values()))`. Once this block goes up. `list(set(encodings.values()))` would have to be tweaked to extract value > shorthand > field